### PR TITLE
Update example and tests for simple-linked-list

### PR DESCRIPTION
**Pull request by matthewmorgan**
This exercise was implemented in a way that really deviated from the README in the problem-specifications repo.

The README is very simple in its requirements:
1. Implement List and Element classes.
2. Implement the List class so that it can:
a) be reversed
b) return an array of element values
c) accept an array as initial data

There's nothing about `pop`, `shift`, etc. The existing implementation is relatively complicated compared to these requirements, as is the test suite, which requires validation that isn't specified in the metadata.

This exercise is already a bit of a mind-bender without all the extra complications-- let's make it simpler by focusing on the core problem.  There are two other exercises that focus on list implementations, so I think we're pretty well covered here.

### CODE REVIEW

**Comment by tejasbubane** [comment]
Do we have to expose `Element` class to users in tests? Can we have `list.add(1)` directly? 
I think that will give users flexibility to have some other way of implementing the exercise. eg. Instead of `class`, one could use plain JS object with two properties `value` and `next` - or even arrays for that matter, though I would not treat the later as linked list.

File: exercises/simple-linked-list/example.js [hunk 1]
```diff
@@ -1,111 +1,52 @@
...
-
-    if (!penultEl) {
-      this.head = undefined;
-    } else {
-      penultEl.next = undefined;
+  countElements(c, element) {
+    let count = c;
+    while (element.next) {
+      count += 1;
+      return this.countElements(count, element.next);
```
Comment (tejasbubane) [inline exercises/simple-linked-list/example.js]: You can use either one of loop or recursion. Just like you did with `toArray`. A return condition before recursion call should work.

**Comment by matthewmorgan** [comment]
@tejasbubane I exposed the `Element` class and used it in tests because the README requires implementing it.

Normally I would agree that this implementation detail should be hidden from consumers of `List`, but because of both the limited scope of the exercise and the explicit call out of the requirement in the README, I thought it should be like this.

Does that make sense?

File: exercises/simple-linked-list/example.js (hunk 1 above)
Comment (matthewmorgan) [reply exercises/simple-linked-list/example.js]: Yes!  I think `reverse` can be recursive also.

I'm going to make those changes as soon as I have a few minutes.

**Comment by tejasbubane** [comment]
Ok it makes sense after reading the README. 👍

**Pull request merged**